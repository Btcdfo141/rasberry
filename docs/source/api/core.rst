.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: