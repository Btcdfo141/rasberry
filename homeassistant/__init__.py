"""Init file for Home Assistant."""


test2
