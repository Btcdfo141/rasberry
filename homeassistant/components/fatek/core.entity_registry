{
  "version": 1,
  "minor_version": 8,
  "key": "core.entity_registry",
  "data": {
    "entities": [
      {
        "area_id": null,
        "capabilities": null,
        "config_entry_id": null,
        "device_class": null,
        "device_id": null,
        "disabled_by": null,
        "entity_category": null,
        "entity_id": "person.marcin_ziarko",
        "hidden_by": null,
        "icon": null,
        "id": "0fa03fbe073a63325a3af0ae10563f01",
        "has_entity_name": false,
        "name": null,
        "options": {},
        "original_device_class": null,
        "original_icon": null,
        "original_name": "Marcin Ziarko",
        "platform": "person",
        "supported_features": 0,
        "unique_id": "marcin_ziarko",
        "unit_of_measurement": null
      },
      {
        "area_id": null,
        "capabilities": {
          "supported_color_modes": [
            "onoff"
          ]
        },
        "config_entry_id": null,
        "device_class": null,
        "device_id": null,
        "disabled_by": null,
        "entity_category": null,
        "entity_id": "light.m878",
        "hidden_by": null,
        "icon": null,
        "id": "379ff55d05771bf72f8ac457aa778ea3",
        "has_entity_name": false,
        "name": null,
        "options": {},
        "original_device_class": null,
        "original_icon": null,
        "original_name": "Poddasze TV",
        "platform": "fatek",
        "supported_features": 0,
        "unique_id": "M878",
        "unit_of_measurement": null
      }
    ]
  }
}
