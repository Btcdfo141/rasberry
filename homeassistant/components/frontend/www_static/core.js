!(function(){"use strict";function t(t){return{type:"auth",api_password:t}}function e(){return{type:"get_states"}}function n(){return{type:"get_config"}}function r(){return{type:"get_services"}}function o(){return{type:"get_panels"}}function i(t,e,n){var r={type:"call_service",domain:t,service:e};return n&&(r.service_data=n),r}function u(t){var e={type:"subscribe_events"};return t&&(e.event_type=t),e}function s(t){return{type:"unsubscribe_events",subscription:t}}function c(){return{type:"ping"}}function a(t,e){return{type:"result",success:!1,error:{code:t,message:e}}}function l(e,n){function r(o,i,u){var s=new WebSocket(e),c=!1,a=function(){if(c)return void u(I);if(0===o)return void u(A);var t=o===-1?-1:o-1;setTimeout((function(){return r(t,i,u)}),1e3)},l=function(e){var r=JSON.parse(e.data);switch(r.type){case"auth_required":"authToken"in n?s.send(JSON.stringify(t(n.authToken))):(c=!0,s.close());break;case"auth_invalid":c=!0,s.close();break;case"auth_ok":s.removeEventListener("message",l),s.removeEventListener("close",a),i(s)}};s.addEventListener("message",l),s.addEventListener("close",a)}return new Promise(function(t,e){return r(n.setupRetry||0,t,e)})}function f(t){return t.result}function p(t,e){return void 0===e&&(e={}),l(t,e).then((function(n){var r=new R(t,e);return r.setSocket(n),r}))}function h(t,e){return t._subscribeConfig?t._subscribeConfig(e):new Promise(function(n,r){function o(t){a=Object.assign({},a,t);for(var e=0;e<f.length;e++)f[e](a)}function i(t){if(null!==a){var e=Object.assign({},a.core,{components:a.core.components.concat(t.data.component)});o({core:e})}}function u(t){if(null!==a){var e,n=t.data,r=n.domain,i=n.service,u=Object.assign({},a.services[r]||{},(e={},e[i]={description:"",fields:{}},e));o({services:Object.assign({},a.services,(s={},s[r]=u,s))});var s}}function s(){return Promise.all([t.getConfig(),t.getPanels(),t.getServices()]).then((function(t){var e=t[0],n=t[1],r=t[2];o({core:e,panels:n,services:r})}))}function c(t){t&&f.splice(f.indexOf(t),1),0===f.length&&l()}var a=null,l=null,f=[],p=null;e&&f.push(e),t._subscribeConfig=function(t){return t&&(f.push(t),null!==a&&t(a)),p.then((function(){return function(){return c(t)}}))},p=Promise.all([t.subscribeEvents(i,"component_loaded"),t.subscribeEvents(u,"service_registered"),s()]),p.then((function(r){var o=r[0],i=r[1];l=function(){removeEventListener("ready",s),o(),i()},t.addEventListener("ready",s),n((function(){return c(e)}))}),(function(){return r()}))})}function d(t){for(var e={},n=0;n<t.length;n++){var r=t[n];e[r.entity_id]=r}return e}function g(t,e){var n=Object.assign({},t);return n[e.entity_id]=e,n}function y(t,e){var n=Object.assign({},t);return delete n[e],n}function v(t,e){return t._subscribeEntities?t._subscribeEntities(e):new Promise(function(n,r){function o(t){if(null!==s){var e=t.data,n=e.entity_id,r=e.new_state;s=r?g(s,r):y(s,n);for(var o=0;o<a.length;o++)a[o](s)}}function i(){return t.getStates().then((function(t){s=d(t);for(var e=0;e<a.length;e++)a[e](s)}))}function u(e){e&&a.splice(a.indexOf(e),1),0===a.length&&(c(),t.removeEventListener("ready",i),t._subscribeEntities=null)}var s=null,c=null,a=[],l=null;e&&a.push(e),t._subscribeEntities=function(t){return t&&(a.push(t),null!==s&&t(s)),l.then((function(){return function(){return u(t)}}))},l=Promise.all([t.subscribeEvents(o,"state_changed"),i()]),l.then((function(r){var o=r[0];c=o,t.addEventListener("ready",i),n((function(){return u(e)}))}),(function(){return r()}))})}function b(t){return t.substr(0,t.indexOf("."))}function m(t){return t.substr(t.indexOf(".")+1)}function w(t,e){var n={};return e.attributes.entity_id.forEach((function(e){var r=t[e];r&&(n[r.entity_id]=r)})),n}function j(t){var e=[],n={};return Object.keys(t).forEach((function(r){var o=t[r];"group"===b(r)?e.push(o):n[r]=o})),e.sort((function(t,e){return t.attributes.order-e.attributes.order})),e.forEach((function(t){return t.attributes.entity_id.forEach((function(t){delete n[t]}))})),{groups:e,ungrouped:n}}function O(t,e){var n={};return e.attributes.entity_id.forEach((function(e){var r=t[e];if(r&&!r.attributes.hidden&&(n[r.entity_id]=r,"group"===b(r.entity_id))){var o=w(t,r);Object.keys(o).forEach((function(t){var e=o[t];e.attributes.hidden||(n[t]=e)}))}})),n}function S(t){var e=[];return Object.keys(t).forEach((function(n){var r=t[n];r.attributes.view&&e.push(r)})),e.sort((function(t,e){return t.entity_id===W?-1:e.entity_id===W?1:t.attributes.order-e.attributes.order})),e}function E(t){var e=V.call(t);return"[object Function]"===e||"function"==typeof t&&"[object RegExp]"!==e||"undefined"!=typeof window&&(t===window.setTimeout||t===window.alert||t===window.confirm||t===window.prompt)}function _(t,e,n){if(!q(e))throw new TypeError("iterator must be a function");arguments.length<3&&(n=this),"[object Array]"===U.call(t)?k(t,e,n):"string"==typeof t?T(t,e,n):P(t,e,n)}function k(t,e,n){for(var r=0,o=t.length;r<o;r++)z.call(t,r)&&e.call(n,t[r],r,t)}function T(t,e,n){for(var r=0,o=t.length;r<o;r++)e.call(n,t.charAt(r),r,t)}function P(t,e,n){for(var r in t)z.call(t,r)&&e.call(n,t[r],r,t)}function C(t){var e={};return Pe(t,(function(t,n){Pe(t,(function(t){e[t]=n}))})),e}function x(t){var e=C(Ie);return e[t]||e[Ne.call(t,/-/,1)[0]]||e.en}function L(t,e){return Ae[x(t)](e)}function M(t,e,n){if("string"!=typeof t)throw new TypeError("Polyglot.transformPhrase expects argument #1 to be string");if(null==e)return t;var r=t,o="number"==typeof e?{smart_count:e}:e;if(null!=o.smart_count&&r){var i=Ne.call(r,$e);r=Le(i[L(n||"en",o.smart_count)]||i[0])}return r=Fe.call(r,We,(function(t,e){return xe(o,e)&&null!=o[e]?Fe.call(o[e],De,Re):t}))}function F(t){var e=t||{};this.phrases={},this.extend(e.phrases||{}),this.currentLocale=e.locale||"en";var n=e.allowMissing?M:null;this.onMissingKey="function"==typeof e.onMissingKey?e.onMissingKey:n,this.warn=e.warn||Me}function N(){function t(t){for(var n=t.toLowerCase().split("-"),r=n.length;r>=1;r--){var o=n.slice(0,r).join("-");if(e[o])return e[o]}return null}var e={};Object.keys(He).forEach((function(t){e[t.toLowerCase()]=t}));var n=null;if(window.localStorage.language){if(n=t(window.localStorage.language))return n}else if(navigator.languages){for(var r=0;r<navigator.languages.length;r++)if(n=t(navigator.languages[r]))return n}else if(n=t(navigator.language||navigator.userLanguage))return n;return"en"}function $(){var t=N(),e=t+"-"+He[t];return qe[e]?new Promise(function(t,n){t(qe[e])}):new Promise(function(n,r){var o,o="/static/translations/"+e+".json";fetch(o,{}).then((function(t){return t.json()})).then((function(r){qe[e]=new Be({phrases:r,locale:t}),n(qe[e])})).catch((function(e){console.error("Unable to load translation "+t+". Falling back to en."),n(new Be({phrases:Ve,locale:"en"}))}))})}var A=1,I=2,D=3,R=function(t,e){this.url=t,this.options=e||{},this.commandId=1,this.commands={},this.eventListeners={},this.closeRequested=!1,this._handleMessage=this._handleMessage.bind(this),this._handleClose=this._handleClose.bind(this)};R.prototype.setSocket=function(t){var e=this,n=this.socket;if(this.socket=t,t.addEventListener("message",this._handleMessage),t.addEventListener("close",this._handleClose),n){var r=this.commands;this.commandId=1,this.commands={},Object.keys(r).forEach((function(t){var n=r[t];n.eventType&&e.subscribeEvents(n.eventCallback,n.eventType).then((function(t){n.unsubscribe=t}))})),this.fireEvent("ready")}},R.prototype.addEventListener=function(t,e){var n=this.eventListeners[t];n||(n=this.eventListeners[t]=[]),n.push(e)},R.prototype.removeEventListener=function(t,e){var n=this.eventListeners[t];if(n){var r=n.indexOf(e);r!==-1&&n.splice(r,1)}},R.prototype.fireEvent=function(t){var e=this;(this.eventListeners[t]||[]).forEach((function(t){return t(e)}))},R.prototype.close=function(){this.closeRequested=!0,this.socket.close()},R.prototype.getStates=function(){return this.sendMessagePromise(e()).then(f)},R.prototype.getServices=function(){return this.sendMessagePromise(r()).then(f)},R.prototype.getPanels=function(){return this.sendMessagePromise(o()).then(f)},R.prototype.getConfig=function(){return this.sendMessagePromise(n()).then(f)},R.prototype.callService=function(t,e,n){return this.sendMessagePromise(i(t,e,n))},R.prototype.subscribeEvents=function(t,e){var n=this;return this.sendMessagePromise(u(e)).then((function(r){var o={eventCallback:t,eventType:e,unsubscribe:function(){return n.sendMessagePromise(s(r.id)).then((function(){delete n.commands[r.id]}))}};return n.commands[r.id]=o,function(){return o.unsubscribe()}}))},R.prototype.ping=function(){return this.sendMessagePromise(c())},R.prototype.sendMessage=function(t){this.socket.send(JSON.stringify(t))},R.prototype.sendMessagePromise=function(t){var e=this;return new Promise(function(n,r){e.commandId+=1;var o=e.commandId;t.id=o,e.commands[o]={resolve:n,reject:r},e.sendMessage(t)})},R.prototype._handleMessage=function(t){var e=JSON.parse(t.data);switch(e.type){case"event":this.commands[e.id].eventCallback(e.event);break;case"result":e.success?this.commands[e.id].resolve(e):this.commands[e.id].reject(e.error),delete this.commands[e.id];break;case"pong":}},R.prototype._handleClose=function(){var t=this;if(Object.keys(this.commands).forEach((function(e){var n=t.commands[e],r=n.reject;r&&r(a(D,"Connection lost"))})),!this.closeRequested){this.fireEvent("disconnected");var e=Object.assign({},this.options,{setupRetry:0}),n=function(r){setTimeout((function(){l(t.url,e).then((function(e){return t.setSocket(e)}),(function(){return n(r+1)}))}),1e3*Math.min(r,5))};n(0)}};var W="group.default_view",B=Object.freeze({ERR_CANNOT_CONNECT:A,ERR_INVALID_AUTH:I,createConnection:p,subscribeConfig:h,subscribeEntities:v,getGroupEntities:w,splitByGroups:j,getViewEntities:O,extractViews:S,extractDomain:b,extractObjectId:m}),H=E,V=Object.prototype.toString,q=H,K=_,U=Object.prototype.toString,z=Object.prototype.hasOwnProperty,J=!1,G=function(){};J&&(G=function(t,e,n){var r=arguments,o=arguments.length;n=new Array(o>2?o-2:0);for(var i=2;i<o;i++)n[i-2]=r[i];if(void 0===e)throw new Error("`warning(condition, format, ...args)` requires a warning message argument");if(e.length<10||/^[s\W]*$/.test(e))throw new Error("The warning format should be able to uniquely identify this warning. Please, use a more descriptive format than: "+e);if(!t){var u=0,s="Warning: "+e.replace(/%s/g,(function(){return n[u++]}));"undefined"!=typeof console&&console.error(s);try{throw new Error(s)}catch(t){}}});var X=G,Y="Function.prototype.bind called on incompatible ",Q=Array.prototype.slice,Z=Object.prototype.toString,tt="[object Function]",et=function(t){var e=this;if("function"!=typeof e||Z.call(e)!==tt)throw new TypeError(Y+e);for(var n,r=Q.call(arguments,1),o=function(){if(this instanceof n){var o=e.apply(this,r.concat(Q.call(arguments)));return Object(o)===o?o:this}return e.apply(t,r.concat(Q.call(arguments)))},i=Math.max(0,e.length-r.length),u=[],s=0;s<i;s++)u.push("$"+s);if(n=Function("binder","return function ("+u.join(",")+"){ return binder.apply(this,arguments); }")(o),e.prototype){var c=function(){};c.prototype=e.prototype,n.prototype=new c,c.prototype=null}return n},nt=et,rt=Function.prototype.bind||nt,ot=rt,it=ot.call(Function.call,Object.prototype.hasOwnProperty),ut=Object.prototype.toString,st=function(t){var e=ut.call(t),n="[object Arguments]"===e;return n||(n="[object Array]"!==e&&null!==t&&"object"==typeof t&&"number"==typeof t.length&&t.length>=0&&"[object Function]"===ut.call(t.callee)),n},ct=Object.prototype.hasOwnProperty,at=Object.prototype.toString,lt=Array.prototype.slice,ft=st,pt=Object.prototype.propertyIsEnumerable,ht=!pt.call({toString:null},"toString"),dt=pt.call((function(){}),"prototype"),gt=["toString","toLocaleString","valueOf","hasOwnProperty","isPrototypeOf","propertyIsEnumerable","constructor"],yt=function(t){var e=t.constructor;return e&&e.prototype===t},vt={$console:!0,$external:!0,$frame:!0,$frameElement:!0,$frames:!0,$innerHeight:!0,$innerWidth:!0,$outerHeight:!0,$outerWidth:!0,$pageXOffset:!0,$pageYOffset:!0,$parent:!0,$scrollLeft:!0,$scrollTop:!0,$scrollX:!0,$scrollY:!0,$self:!0,$webkitIndexedDB:!0,$webkitStorageInfo:!0,$window:!0},bt=(function(){if("undefined"==typeof window)return!1;for(var t in window)try{if(!vt["$"+t]&&ct.call(window,t)&&null!==window[t]&&"object"==typeof window[t])try{yt(window[t])}catch(t){return!0}}catch(t){return!0}return!1})(),mt=function(t){if("undefined"==typeof window||!bt)return yt(t);try{return yt(t)}catch(t){return!1}},wt=function(t){var e=null!==t&&"object"==typeof t,n="[object Function]"===at.call(t),r=ft(t),o=e&&"[object String]"===at.call(t),i=[];if(!e&&!n&&!r)throw new TypeError("Object.keys called on a non-object");var u=dt&&n;if(o&&t.length>0&&!ct.call(t,0))for(var s=0;s<t.length;++s)i.push(String(s));if(r&&t.length>0)for(var c=0;c<t.length;++c)i.push(String(c));else for(var a in t)u&&"prototype"===a||!ct.call(t,a)||i.push(String(a));if(ht)for(var l=mt(t),f=0;f<gt.length;++f)l&&"constructor"===gt[f]||!ct.call(t,gt[f])||i.push(gt[f]);return i};wt.shim=function(){if(Object.keys){var t=(function(){return 2===(Object.keys(arguments)||"").length})(1,2);if(!t){var e=Object.keys;Object.keys=function(t){return e(ft(t)?lt.call(t):t)}}}else Object.keys=wt;return Object.keys||wt};var jt=wt,Ot=Object.prototype.hasOwnProperty,St=Object.prototype.toString,Et=function(t,e,n){if("[object Function]"!==St.call(e))throw new TypeError("iterator must be a function");var r=t.length;if(r===+r)for(var o=0;o<r;o++)e.call(n,t[o],o,t);else for(var i in t)Ot.call(t,i)&&e.call(n,t[i],i,t)},_t=jt,kt=Et,Tt="function"==typeof Symbol&&"symbol"==typeof Symbol(),Pt=Object.prototype.toString,Ct=function(t){return"function"==typeof t&&"[object Function]"===Pt.call(t)},xt=function(){var t={};try{Object.defineProperty(t,"x",{enumerable:!1,value:t});for(var e in t)return!1;return t.x===t}catch(t){return!1}},Lt=Object.defineProperty&&xt(),Mt=function(t,e,n,r){(!(e in t)||Ct(r)&&r())&&(Lt?Object.defineProperty(t,e,{configurable:!0,enumerable:!1,value:n,writable:!0}):t[e]=n)},Ft=function(t,e){var n=arguments.length>2?arguments[2]:{},r=_t(e);Tt&&(r=r.concat(Object.getOwnPropertySymbols(e))),kt(r,(function(r){Mt(t,r,e[r],n[r])}))};Ft.supportsDescriptors=!!Lt;var Nt=Ft,$t=Number.isNaN||function(t){return t!==t},At=Number.isNaN||function(t){return t!==t},It=Number.isFinite||function(t){return"number"==typeof t&&!At(t)&&t!==1/0&&t!==-(1/0)},Dt=function(t){return t>=0?1:-1},Rt=function(t,e){var n=t%e;return Math.floor(n>=0?n:n+e)},Wt=Function.prototype.toString,Bt=/^\s*class /,Ht=function(t){try{var e=Wt.call(t),n=e.replace(/\/\/.*\n/g,""),r=n.replace(/\/\*[.\s\S]*\*\//g,""),o=r.replace(/\n/gm," ").replace(/ {2}/g," ");return Bt.test(o)}catch(t){return!1}},Vt=function(t){try{return!Ht(t)&&(Wt.call(t),!0)}catch(t){return!1}},qt=Object.prototype.toString,Kt="[object Function]",Ut="[object GeneratorFunction]",zt="function"==typeof Symbol&&"symbol"==typeof Symbol.toStringTag,Jt=function(t){if(!t)return!1;if("function"!=typeof t&&"object"!=typeof t)return!1;if(zt)return Vt(t);if(Ht(t))return!1;var e=qt.call(t);return e===Kt||e===Ut},Gt=function(t){return null===t||"function"!=typeof t&&"object"!=typeof t},Xt=Object.prototype.toString,Yt=Gt,Qt=Jt,Zt={"[[DefaultValue]]":function(t,e){var n=e||("[object Date]"===Xt.call(t)?String:Number);if(n===String||n===Number){var r,o,i=n===String?["toString","valueOf"]:["valueOf","toString"];for(o=0;o<i.length;++o)if(Qt(t[i[o]])&&(r=t[i[o]](),Yt(r)))return r;throw new TypeError("No default value")}throw new TypeError("invalid [[DefaultValue]] hint supplied")}},te=function(t,e){return Yt(t)?t:Zt["[[DefaultValue]]"](t,e)},ee=$t,ne=It,re=Dt,oe=Rt,ie=Jt,ue=te,se={ToPrimitive:ue,ToBoolean:function(t){return Boolean(t)},ToNumber:function(t){return Number(t)},ToInteger:function(t){var e=this.ToNumber(t);return ee(e)?0:0!==e&&ne(e)?re(e)*Math.floor(Math.abs(e)):e},ToInt32:function(t){return this.ToNumber(t)>>0},ToUint32:function(t){return this.ToNumber(t)>>>0},ToUint16:function(t){var e=this.ToNumber(t);if(ee(e)||0===e||!ne(e))return 0;var n=re(e)*Math.floor(Math.abs(e));return oe(n,65536)},ToString:function(t){return String(t)},ToObject:function(t){return this.CheckObjectCoercible(t),Object(t)},CheckObjectCoercible:function(t,e){if(null==t)throw new TypeError(e||"Cannot call method on "+t);return t},IsCallable:ie,SameValue:function(t,e){return t===e?0!==t||1/t===1/e:ee(t)&&ee(e)},Type:function(t){return null===t?"Null":"undefined"==typeof t?"Undefined":"function"==typeof t||"object"==typeof t?"Object":"number"==typeof t?"Number":"boolean"==typeof t?"Boolean":"string"==typeof t?"String":void 0}},ce=se,ae=rt,le=ce,fe=ae.call(Function.call,String.prototype.replace),pe=/^[\x09\x0A\x0B\x0C\x0D\x20\xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u3000\u2028\u2029\uFEFF]+/,he=/[\x09\x0A\x0B\x0C\x0D\x20\xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u3000\u2028\u2029\uFEFF]+$/,de=function(){var t=le.ToString(le.CheckObjectCoercible(this));return fe(fe(t,pe,""),he,"")},ge=de,ye="​",ve=function(){return String.prototype.trim&&ye.trim()===ye?String.prototype.trim:ge},be=Nt,me=ve,we=function(){var t=me();return be(String.prototype,{trim:t},{trim:function(){return String.prototype.trim!==t}}),t},je=rt,Oe=Nt,Se=de,Ee=ve,_e=we,ke=je.call(Function.call,Ee());Oe(ke,{getPolyfill:Ee,implementation:Se,shim:_e});var Te=ke,Pe=K,Ce=X,xe=it,Le=Te,Me=function(t){Ce(!1,t)},Fe=String.prototype.replace,Ne=String.prototype.split,$e="||||",Ae={arabic:function(t){return t<3?t:t%100>=3&&t%100<=10?3:t%100>=11?4:5},chinese:function(){return 0},german:function(t){return 1!==t?1:0},french:function(t){return t>1?1:0},russian:function(t){return t%10===1&&t%100!==11?0:t%10>=2&&t%10<=4&&(t%100<10||t%100>=20)?1:2},czech:function(t){return 1===t?0:t>=2&&t<=4?1:2},polish:function(t){return 1===t?0:t%10>=2&&t%10<=4&&(t%100<10||t%100>=20)?1:2},icelandic:function(t){return t%10!==1||t%100===11?1:0}},Ie={arabic:["ar"],chinese:["fa","id","ja","ko","lo","ms","th","tr","zh"],german:["da","de","en","es","fi","el","he","hu","it","nl","no","pt","sv"],french:["fr","tl","pt-br"],russian:["hr","ru","lt"],czech:["cs","sk"],polish:["pl"],icelandic:["is"]},De=/\$/g,Re="$$",We=/%\{(.*?)\}/g;F.prototype.locale=function(t){return t&&(this.currentLocale=t),this.currentLocale},F.prototype.extend=function(t,e){Pe(t,(function(t,n){var r=e?e+"."+n:n;"object"==typeof t?this.extend(t,r):this.phrases[r]=t}),this)},F.prototype.unset=function(t,e){"string"==typeof t?delete this.phrases[t]:Pe(t,(function(t,n){var r=e?e+"."+n:n;"object"==typeof t?this.unset(t,r):delete this.phrases[r]}),this)},F.prototype.clear=function(){this.phrases={}},F.prototype.replace=function(t){this.clear(),this.extend(t)},F.prototype.t=function(t,e){var n,r,o=null==e?{}:e;if("string"==typeof this.phrases[t])n=this.phrases[t];else if("string"==typeof o._)n=o._;else if(this.onMissingKey){var i=this.onMissingKey;r=i(t,o,this.currentLocale)}else this.warn('Missing translation for key: "'+t+'"'),r=t;return"string"==typeof n&&(r=M(n,o,this.currentLocale)),r},F.prototype.has=function(t){return xe(this.phrases,t)},F.transformPhrase=M;var Be=F,He={en:"86e1bc436d00e74f02f5ae6c2c994e4d"},Ve={panel:{states:"States",map:"Map",logbook:"Logbook",history:"History",log_out:"Log Out"}},qe={};$(),window.HAWS=B,window.HASS_DEMO=!1,window.getPolyglot=$;var Ke=window.createHassConnection=function(t){var e="https:"===window.location.protocol?"wss":"ws",n=e+"://"+window.location.host+"/api/websocket",r={setupRetry:10};return void 0!==t&&(r.authToken=t),p(n,r).then((function(t){return v(t),h(t),t}))};window.noAuth?window.hassConnection=Ke():window.localStorage.authToken?window.hassConnection=Ke(window.localStorage.authToken):window.hassConnection=null,"serviceWorker"in navigator&&window.addEventListener("load",(function(){navigator.serviceWorker.register("/service_worker.js")}))})();
