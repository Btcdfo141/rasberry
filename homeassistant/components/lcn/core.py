"""Core dependencies for LCN component."""

import re

import voluptuous as vol

# definitions
CONF_SK_NUM_TRIES = 'sk_num_tries'
CONF_DIM_MODE = 'dim_mode'
CONF_OUTPUT = 'output'
CONF_TRANSITION = 'transition'
CONF_DIMMABLE = 'dimmable'
CONF_CONNECTIONS = 'connections'

DIM_MODES = ['steps50', 'steps200']
OUTPUT_PORTS = ['output1', 'output2', 'output3', 'output4']

# Regex for address validation
PATTERN_ADDRESS = re.compile('^((?P<conn_id>\\w+)\\.)?s?(?P<seg_id>\\d+)'
                             '\\.(?P<type>m|g)?(?P<id>\\d+)$')


def get_connection(connections, connection_id=None):
    """Return the connection object from list."""
    if connection_id is None:
        connection = connections[0]
    else:
        for connection in connections:
            if connection.connection_id == connection_id:
                break
        else:
            raise ValueError('Unknown connection_id.')
    return connection


def is_address(value):
    """Validate the given address string.

    Examples for S000M005 at myhome:
        myhome.s000.m005
        myhome.s0.m5
        myhome.0.5    ("m" is implicit if missing)

    Examples for s000g011
        myhome.0.g11
        myhome.s0.g11
    """
    matcher = PATTERN_ADDRESS.match(value)
    if matcher:
        is_group = (matcher.group('type') == 'g')
        addr = (int(matcher.group('seg_id')),
                int(matcher.group('id')),
                is_group)
        conn_id = matcher.group('conn_id')
        return addr, conn_id
    raise vol.error.Invalid('Not a valid address string.')
