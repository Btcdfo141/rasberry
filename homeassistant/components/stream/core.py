"""Provides core stream functionality."""
import asyncio
import io
from typing import List, Any

import attr
from aiohttp import web

from homeassistant.components.http import HomeAssistantView
from homeassistant.helpers.event import async_call_later

from .const import DOMAIN, ATTR_STREAMS


@attr.s
class StreamBuffer:
    """Represent a segment."""

    segment = attr.ib(type=io.BytesIO)
    output = attr.ib()               # type=av.OutputContainer
    vstream = attr.ib()              # type=av.VideoStream
    astream = attr.ib(default=None)  # type=av.AudioStream


@attr.s
class Segment:
    """Represent a segment."""

    sequence = attr.ib(type=int)
    segment = attr.ib(type=io.BytesIO)
    duration = attr.ib(type=float)


class StreamOutput:
    """Represents a stream output."""

    num_segments = 3

    def __init__(self) -> None:
        """Initialize a stream output."""
        self._cursor = None
        self._event = asyncio.Event()
        self._segments = []

    @property
    def format(self) -> str:
        """Return container format."""
        return None

    @property
    def audio_codec(self) -> str:
        """Return desired audio codec."""
        return None

    @property
    def video_codec(self) -> str:
        """Return desired video codec."""
        return None

    @property
    def segments(self) -> List[int]:
        """Return current sequence from segments."""
        return [s.sequence for s in self._segments]

    @property
    def target_duration(self) -> int:
        """Return the average duration of the segments in seconds."""
        durations = [s.duration for s in self._segments]
        return round(sum(durations) // len(self._segments)) or 1

    def get_segment(self, sequence: int = None) -> Any:
        """Retrieve a specific segment, or the whole list."""
        if not sequence:
            return self._segments

        for segment in self._segments:
            if segment.sequence == sequence:
                return segment
        return None

    async def recv(self) -> Segment:
        """Wait for and retrieve the latest segment."""
        last_segment = max(self.segments, default=0)
        if self._cursor is None or self._cursor <= last_segment:
            await self._event.wait()

        if not self._segments:
            return None

        segment = self._segments[-1]
        self._cursor = segment.sequence
        return segment

    async def put(self, segment: Segment) -> None:
        """Store output."""
        if segment is None:
            self._segments = []
            self._event.set()
            return

        self._segments.append(segment)
        if len(self._segments) > self.num_segments:
            self._segments = self._segments[-self.num_segments:]
        self._event.set()
        self._event.clear()


class StreamView(HomeAssistantView):
    """
    Base StreamView.

    For implementation of a new stream format, define `url` and `name`
    attributes, and implement `handle` method in a child class.
    """

    requires_auth = False
    platform = None

    def __init__(self):
        """Initialize a basic stream view."""
        self._unsub = None

    async def get(self, request, token, sequence=None):
        """Start a GET request."""
        hass = request.app['hass']

        stream = next((
            s for s in hass.data[DOMAIN][ATTR_STREAMS].values()
            if s.access_token == token), None)

        if not stream:
            raise web.HTTPNotFound()

        # Start worker if not already started
        stream.start()

        if self._unsub is not None:
            self._unsub()

        async def cleanup(_now):
            """Stop the stream."""
            stream.stop()
            self._unsub = None

        self._unsub = async_call_later(hass, 30, cleanup)

        return await self.handle(request, stream, sequence)

    async def handle(self, request, stream, sequence):
        """Handle the stream request."""
        raise NotImplementedError()
