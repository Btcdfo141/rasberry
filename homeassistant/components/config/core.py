"""Component to interact with Hassbian tools."""
import asyncio

from homeassistant.components.http import HomeAssistantView
from homeassistant.config import async_check_ha_config_file


@asyncio.coroutine
def async_setup(hass):
    """Set up the Hassbian config."""
    hass.http.register_view(CheckConfigView)
    return True


class CheckConfigView(HomeAssistantView):
    """Hassbian packages endpoint."""

    url = '/api/config/core/check_config'
    name = 'api:config:core:check_config'

    @asyncio.coroutine
    def post(self, request):
        """Validate configuration and return results."""
        hass = request.app['hass']

        if hass.components.hassio.is_hassio():
            errors = yield from hass.components.hassio.async_check_config()
        else:
            errors = yield from async_check_ha_config_file(hass)

        state = 'invalid' if errors else 'valid'

        return self.json({
            "result": state,
            "errors": errors,
        })
