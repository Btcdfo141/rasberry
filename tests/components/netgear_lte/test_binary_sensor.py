"""The tests for Netgear LTE binary sensor platform."""
from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant


async def test_binary_sensors(
    hass: HomeAssistant,
    entity_registry_enabled_by_default: None,
    setup_integration: None,
    snapshot: SnapshotAssertion,
) -> None:
    """Test for successfully setting up the Netgear LTE binary sensor platform."""
    assert hass.states.get("binary_sensor.netgear_lm1200_mobile_connected") == snapshot

    assert hass.states.get("binary_sensor.netgear_lm1200_wire_connected") == snapshot

    assert hass.states.get("binary_sensor.netgear_lm1200_roaming") == snapshot
