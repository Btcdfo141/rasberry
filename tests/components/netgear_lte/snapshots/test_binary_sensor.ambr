# serializer version: 1
# name: test_binary_sensors
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'connectivity',
      'friendly_name': 'Netgear LM1200 Mobile connected',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.netgear_lm1200_mobile_connected',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_binary_sensors.1
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'connectivity',
      'friendly_name': 'Netgear LM1200 Wire connected',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.netgear_lm1200_wire_connected',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors.2
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Netgear LM1200 Roaming',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.netgear_lm1200_roaming',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
